pyproject.toml

```diff
@@ -50,7 +50,44 @@ disallow_any_decorated = true
 
 exclude = [
   "typeshed_client/typeshed",
+  "tests/site-packages",
   "tests/typeshed",
   "build/",
+  "thirdparty",
   ".tox/",
 ]
+
+[tool.ruff]
+
+line-length = 100
+target-version = "py38"
+preview = true
+unsafe-fixes = true
+
+exclude = [
+  "typeshed_client/typeshed",
+  "tests/typeshed",
+  "build/",
+  ".tox/",
+]
+
+[tool.ruff.lint]
+select = [
+  "F",
+  "E",
+  "I",  # import sorting
+  "ANN",  # enforce type annotations
+  "C4",  # flake8-comprehensions
+  "B",  # bugbear
+  "SIM",  # simplify
+  "UP",  # pyupgrade
+  "PIE",
+  "PERF",
+  "RUF",  # Ruff's own rules
+]
+
+ignore = [
+  "ANN101",  # missing type annotation for self in method
+  "ANN102",  # missing type annotation for cls in classmethod
+  "SIM105",  # I don't like contextlib.suppress
```

Comment:
I always ignore `UP038` because of https://github.com/astral-sh/ruff/issues/7871, though it doesn't make much of a difference here since the rule only does things if your lowest supported version is Python 3.10

Reply:
Yeah I agree with you there, adding support for `isinstance()` with UnionType was a questionable design choice in the first place. I'll add it to the exclude list and see if I still agree in two years' time when 3.9 goes EOL :)